src/pms/controller/forecasters/statistical.py

```
@@ -1,23 +1,240 @@
+
+def _confidence(probabilities: Sequence[float], weights: Sequence[float]) -> float:
+    if len(probabilities) == 1:
+        return 0.5
+    total_weight = sum(weights)
+    if total_weight == 0.0:
+        weights = tuple(1.0 for _ in probabilities)
+        total_weight = float(len(probabilities))
+    mean = (
+        sum(
+            probability * weight
+            for probability, weight in zip(probabilities, weights)
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)